### PR TITLE
Update internal dev dependencies for GitHub

### PR DESCRIPTION
**Pull request by AndrewGable**
### Details

As a part of upgrading to NPM 8, we ran into a large peer dependency issue with GitHub dependencies. This upgrades all GitHub dependencies to the latest version and uses the new rest API.

This updates it in the way that @roryabraham and I discussed would be easiest to understand.

### Fixed Issues

Related https://github.com/Expensify/App/issues/10331

### Tests
1. We will need to test this on production by merging this, and verifying none of the deploy process is broken.
2. Additionally, there are some automated tests that will help test this.

### CODE REVIEW

**Comment by AndrewGable** [comment]
Somehow the paginate is very broken now, trying to figure out how to fix it. Once that is fixed we should be in buisness here.

**Comment by AndrewGable** [comment]
Tests should be passing now

**Comment by AndrewGable** [comment]
Ok should see a ✅ now and ready for review!

**Comment by arosiclair** [comment]
I'll defer to Rory since I'm 100% green on github actions

File: .github/libs/GithubUtils.js [hunk 1]
```diff
@@ -60,7 +56,18 @@ class GithubUtils {
                 },
             },
         }));
-        return this.octokitInternal;
+        return this.internalOctokit;
```
[the remark this answers is not in the document]
Comment (AndrewGable) [reply .github/libs/GithubUtils.js:59]: We need both objects: `octokit` (for the `paginate` plugin) and `octokit.rest` for everything else. Happy to refactor but this is the cleanest I could think of with both objects being required.

[the remark this answers is not in the document]
Comment (roryabraham) [reply .github/libs/GithubUtils.js:59]: Ah, I see. Here's a way we could refactor things that I think might be a bit cleaner?

```js
class GitHubUtils {
	/**
	 * Initialize internal octokit
	 * 
	 * @private
	 * @memberof GitHubUtils
	 */
    #initOctokit() {
        const Octokit = GitHub.plugin(throttling, paginateRest);
        const token = core.getInput('GITHUB_TOKEN', {required: true});

        // Save a copy of octokit used in this class
        this.internalOctokit = new Octokit(getOctokitOptions(token, {
            throttle: {
                onRateLimit: (retryAfter, options) => {
                    console.warn(
                        `Request quota exhausted for request ${options.method} ${options.url}`,
                    );
                    // Retry once after hitting a rate limit error, then give up
                    if (options.request.retryCount <= 1) {
                        console.log(`Retrying after ${retryAfter} seconds!`);
                        return true;
                    }
                },
                onAbuseLimit: (retryAfter, options) => {
                    // does not retry, only logs a warning
                    console.warn(
                        `Abuse detected for request ${options.method} ${options.url}`,
                    );
                },
            },
        }));
    }
    
    /**
     * Either give an existing instance of Octokit rest or create a new one
     *
     * @readonly
     * @static
     * @memberof GithubUtils
     */
    static get octokit() {
        if (this.internalOctokit) {
            return this.internalOctokit.rest;
        }
        this.initOctokit();
        return this.internalOctokit.rest;
    }

	/**
     * Either give an existing instance of Octokit paginate or create a new one
     *
     * @readonly
     * @static
     * @memberof GithubUtils
     */
    static get paginate() {
        if (this.internalOctokit) {
            return this.internalOctokit.paginate;
        }
        this.initOctokit();
        return this.internalOctokit.paginate;
    }
}
```

[the remark this answers is not in the document]
Comment (roryabraham) [reply .github/libs/GithubUtils.js:59]: Note that `#initOctokit` is using [private member syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields), and is optional. You could just make it a normal method too like `initOctokit` (I'm just partial to thinking about things in a more object-oriented way)

[the remark this answers is not in the document]
Comment (AndrewGable) [reply .github/libs/GithubUtils.js:59]: Love it! I will update with that soon.

**Comment by AndrewGable** [comment]
Diff is looking much more manageable at this point, thanks for the suggestion!

**Review by roryabraham — APPROVED** [review]

**Pull request merged**